web/Controllers/LandlordController.cs

```diff
@@ -141,4 +142,32 @@ public IActionResult ViewProperties()
         var listOfProperties = databaseResult.Select(PropertyViewModel.FromDbModel).ToList();
         return View("Properties", new PropertiesDashboardViewModel(listOfProperties));
     }
+
+    [HttpGet]
+    [Route("edit")]
+    public async Task<ActionResult> EditProfilePage(int landlordId)
+    {
+        var user = GetCurrentUser();
+        var landlordFromDb =  await _landlordService.GetLandlordIfExistsFromId(landlordId);
+        if (landlordFromDb == null) return StatusCode(404);
+        if (user.LandlordId != landlordFromDb.Id && !user.IsAdmin) return StatusCode(403);
+        return View("EditProfilePage", LandlordProfileModel.FromDbModel(landlordFromDb, user));
+    }
+
+    [HttpPost]
+    [Route("edit")]
+    public async Task<ActionResult> EditProfileUpdate([FromForm] LandlordProfileModel editModel)
+    {
+        var isEmailDuplicated = _landlordService.CheckForDuplicateEmail(editModel);

```

Comment:
If I'm going to check both this and the other edit method using the same conditions, is it worth putting that check into a method since I'd be using it twice? If so, should that method go in the controller or the service?

Reply:
I wrote a comment here but realised I was being silly so have deleted it